lib/ansible/modules/cloud/amazon/ec2_vpc_eigw.py

```diff
@@ -0,0 +1,214 @@
+
+# Ensure that the VPC has an Internet Gateway.
+# The Internet Gateway ID is can be accessed via {{eigw.gateway_id}} for use in setting up NATs etc.
+ec2_vpc_eigw:
+  vpc_id: vpc-abcdefgh
+  state: present
+register: eigw
+
+'''
+
+RETURN = '''
+gateway_id:
+    description: The ID of the Egress Only Internet Gateway
+    returned: success
+    type: string
+    sample: eigw-0e00cf111ba5bc11e
+'''
+
+
+import traceback
+from time import sleep
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    AWSRetry,
+    HAS_BOTO3,
+    boto3_tag_list_to_ansible_dict,
+    camel_dict_to_snake_dict,
+    ansible_dict_to_boto3_filter_list
+)
+from botocore.exceptions import ClientError
+
+
+@AWSRetry.backoff()
+def delete_eigw(module, client, eigw_id):
+    """
+    Delete EIGW
+
+    module     : AnsibleModule object
+    client     : boto3 client connection object
+    eigw_id    : ID of the EIGW to delete
+    """
+
+    try:
+        response = client.delete_egress_only_internet_gateway(DryRun=module.check_mode, EgressOnlyInternetGatewayId=eigw_id)
+    except ClientError as err:
+        # When boto3 method is run with DryRun=True it returns an error on success
+        # We need to catch the error and return something valid
+        if err.response.get('Error').get('Code') == "DryRunOperation":
+            return {'changed': True, 'gateway_id': None}
+        else:
+            module.fail_json(msg=err.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(err.response))
+
+    return {'changed': response['ReturnCode']}
+
+
+@AWSRetry.backoff()
+def create_eigw(module, client, vpc_id):
+    """
+    Create EIGW
+
+    module     : AnsibleModule object
+    client     : boto3 client connection object
+    vpc_id     : ID of the VPC we are operating on
+    """
+
+    try:
+        response = client.create_egress_only_internet_gateway(DryRun=module.check_mode, VpcId=vpc_id)
+        eigw = response['EgressOnlyInternetGateway']
+    except ClientError as err:
+        # When boto3 method is run with DryRun=True it returns an error on success
+        # We need to catch the error and return something valid
+        if err.response.get('Error').get('Code') == "DryRunOperation":
+            return {'changed': True, 'gateway_id': None}
+        else:
+            module.fail_json(msg=err.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(err.response))
+
+    # We loop through the Attachments list in boto3 response to make sure the EIGW is fully attached before returning
+    state = eigw['Attachments'][0]['State']
+    if state == 'attached':
+        # EIGW is fully attached so we return immediately
+        return {'changed': True, 'gateway_id': eigw['EgressOnlyInternetGatewayId']}
+    elif state == 'attaching':
+        # EIGW is still attaching so we check status
```

Comment:
I would consider adding a `wait` and `wait_timeout` attribute to this module and only wait if `wait` is true.